libr/cons/input.c

```diff
@@ -435,6 +435,51 @@ R_API int r_cons_fgets(char *buf, int len, int argc, const char **argv) {
 	return ret;
 }
 
+R_API char *r_cons_get_input(int *status) {
+	RCons *cons = r_cons_singleton ();
+	const int color = cons->context->pal.input && *cons->context->pal.input;
+	if (cons->echo) {
+		r_cons_set_raw (false);
+		r_cons_show_cursor (true);
+	}
+	if (cons->user_get_input) {
+		return cons->user_get_input(status);
+	}
+	printf ("%s", cons->line->prompt);
+	fflush (stdout);
+	if (color) {
+		const char *p = cons->context->pal.input;
+		int len = p? strlen (p): 0;
+		if (len > 0) {
+			fwrite (p, len, 1, stdout);
+		}
+		fflush (stdout);
+	}
+	RStrBuf sb;
+	r_strbuf_init (&sb);
+	char buf[1024];
+	int s = 0;
+	do {
+		buf[1022] = '\0';
```

Comment:
do this *after* the fgets, otherwise the buffer can be non null terminated

Reply:
but actually you can read a raw buffer and pass it to sb.setbin() with buf + size